sys/multi_content_bucket.js

```diff
@@ -428,7 +428,10 @@ class MultiContentBucket {
                         })
                         .then((res) => {
                             if (res.body.items.length) {
-                                return this._deleteRenders(hyper, req, rev, res.body.items[0].tid);
+                                return P.all(
+                                    this._deleteRenders(hyper, req, rev, res.body.items[0].tid),
+                                    this._deleteRevisions(hyper, req, rev - 1)
```

Comment:
> Do we really need to execute that only when res.body.items is not empty?

Absolutely. We can only delete the previous revisions if they were replaced more then TTL ago. Since we don't look into the revision timeline, the only way to ensure that previous revision was replaced more then TTL ago is to ensure that a render that was replaced more then the TTL ago also exist.